dlc-manager/src/lib.rs

```diff
@@ -77,27 +84,75 @@ impl Time for SystemTimeProvider {
 }
 
 /// Provides signing related functionalities.
-pub trait Signer {
-    /// Signs a transaction input
-    fn sign_psbt_input(
-        &self,
-        psbt: &mut PartiallySignedTransaction,
-        input_index: usize,
-    ) -> Result<(), Error>;
+pub trait ContractSigner: Clone {
+    /// Get the public key associated with the [`ContractSigner`].
+    fn get_public_key(&self) -> Result<PublicKey, Error>;
+    /// Returns the secret key associated with the [`ContractSigner`].
+    // todo: remove this method and add create_adaptor_signature to the trait
```

Comment:
I guess it'd need a `sign_cet` or something like that as well to finalize a transaction before broadcasting.